exam_django/asset/service/asset_assign_crud_service/assign_asset_service.py

```diff
@@ -0,0 +1,59 @@
+from response import APIResponse
+from asset.service.asset_assign_crud_service.asset_sysadmin_role_assignasset_service import (
+    AssetSysadminRoleAssignService,
+)
+from asset.service.asset_assign_crud_service.asset_lead_role_assignasset_service import (
+    AssetLeadRoleAssignService,
+)
+from rest_framework import status
+from messages import UNAUTHORIZED_NO_PERMISSION, EMPLOYEE_NOT_FOUND_ERROR
+from asset.models import Employee
+from notification.service.EmailService import EmailService
+from rest_framework.renderers import JSONRenderer
+from asset.serializers import AssignAssetSerializer
+
+
+class AssignAssetService:
+    @staticmethod
+    def assign_asset(requester_role, asset, employee_id, requester):
+        try:
+            employee = Employee.objects.get(id=employee_id)
+        except Employee.DoesNotExist:
+            return APIResponse(
+                message=EMPLOYEE_NOT_FOUND_ERROR,
+                status=status.HTTP_404_NOT_FOUND,
+            )
+        if requester_role == "SYSTEM_ADMIN":
+            message = AssetSysadminRoleAssignService.assign_asset(
+                asset, employee, requester
+            )
+        elif requester_role == "LEAD":
+            message = AssetLeadRoleAssignService.assign_asset(
+                asset, employee, requester
+            )
+        else:
+            # Only return error message and status code for unauthorized roles
+            return {
+                "message": UNAUTHORIZED_NO_PERMISSION,
+                "status": status.HTTP_403_FORBIDDEN,
+            }
```

Comment:
If this else condition is executed, the returned object is not correctly handled in the assign_asset_view. Check line 49 to 52 in assign_asset_view. It would return
```
            return {
                "message": UNAUTHORIZED_NO_PERMISSION,
                "status": status.HTTP_403_FORBIDDEN,
            }
```
and it will be shown as the message.

My suggestion is to use API response here, saying that it is an unauthorized user.
```suggestion
            return APIResponse(
                message=UNAUTHORIZED_NO_PERMISSION,
                status=status.HTTP_403_FORBIDDEN,
            )
```